cognite/extractorutils/base.py

```diff
@@ -210,37 +212,53 @@ def recursive_find_state_store(d: dict[str, Any]) -> StateStoreConfig | None:
 
         Extractor._statestore_singleton = self.state_store
 
-    def _report_success(self) -> None:
-        """
-        Called on a successful exit of the extractor
-        """
+    def _report_run(self, status: str, message: str) -> None:
```

Comment:
Maybe make the status more restricted.
```suggestion
    ReportStatus = Literal["success", "failure"]
    
    def _report_run(self, status: ReportStatus, message: str) -> None:
```

Reply:
This is a good idea, thanks.